### DISCUSSION TITLE
Gonne give this repo a whirl since I, too...

### DISCUSSION THREAD

**Discussion by myndxero** [vladmandic/sdnext#80] [Ideas]
am disgruntled with Auto's lack absences lately. It's fine if he wants to slow down, take breaks, check out completely, but for the love of god, all he had to do was delegate power to a few trustees to approve merges and such! Been waiting to see a repo crop up that was active. I'm not very good at using github, I felt like I looked at the fork list a few times, but nothing appeared active on there.

Anyway, Hello. 

A feature I've been interested in was a means to reconnect to the image preview if you close the browser for whatever reason, or if youy want to view it remotely from another device. I like watching it from time to do time when running a large job and while doing other things. Purple an entertainment type feature I guess.

Thanks, look forward to seeing how this runs and how it develops.

**Comment by vladmandic** [comment]
yes, i'm aware of that specific issue and its annoying me as well. its pretty high on my list, but not an easy one to fix.

**Reply by myndxero** [reply]
👍😎

Looking forward to your branch, once it works on windows haha. I've not messed with WSL, but I could go that route too.

**Reply by vladmandic** [reply]
i use wsl daily, like it more than native windows as dev environment (and don't want to fully switch to linux as windows is more comfortable for anything thats not dev work).

anyhow, i've posted a major update few minutes ago.

**Comment by myndxero** [comment]
Awesome be on shortly to check it out!!

I haven't used Linux in........ A very long time, not as a primary OS nor
as a dual boot etc. I think the last time I was serious with windows was
when Ubuntu was in it's early infancy, portrays my age a little :P

I do have a steam deck so limited interactions with proton is the most I
can ascribe to these days.

**Comment by vladmandic** [comment]
@myndxero i think i figured out how to reconnect browser to active session on browser start!
there are limitations to it, but it should work for a) if you use same browser (so you cant switch from chrome to firefox and expect it to reconnect), b) use text and image generate, other ops are not reconnectable.

**Comment by myndxero** [comment]
Ooohhh did you push that out today? I had some issues where it wouldn't connect to live preview on a fresh load, idr if I updated or not.

Is this across all devices aw long as browser same? I use Brave, for whatever reason, for all my gradio UIs. Don't have on phone, but maybe since chrome based it might work chrome phone to PC brave? /Shrug. 

Might almost be worth making a chome based extension haha.

That's awesome though! I've been deep into control net.

ps - I know im free to change the skin but I like it mostly. it COULD use a touch of contrast maybe of a dark grey to shake up the almost all black. scroll bar so hard to see. new coat of paint vs dark mode autos1111 is nice.

**Comment by vladmandic** [comment]
> Ooohhh did you push that out today? I had some issues where it wouldn't connect to live preview on a fresh load, idr if I updated or not.

reconnect was pushed yesterday afternoon, but as a result it broke live preview - which i've fixed this morning.

> Is this across all devices aw long as browser same?

it works on the same browser, not accross browsers or accross systems - reason is that `task_id` is stored in browser's local-storage, so next time browser opens it knows what to look for. if i store it on server itself, it would work cross-browser and cross-system, but would be a major security issue as different users can  reconnect to session - imaging you're doing something and someone else can now see it live. sure, not an issue for single-user setups, but i don't want to have security issues in general.

> I know im free to change the skin but I like it mostly. it COULD use a touch of contrast maybe of a dark grey to shake up the almost all black. scroll bar so hard to see. new coat of paint vs dark mode autos1111 is nice.

i'm open to suggestions. even better, if you can try editing `javascript/black-orange.css` directly and make suggestions there (if you're comfortable with css). if not, just describe in details in a discussion thread.

**Reply by myndxero** [reply]
> i'm open to suggestions. even better, if you can try editing `javascript/black-orange.css` directly and make suggestions there (if you're comfortable with css). if not, just describe in details in a discussion thread.

---------------
TL;DR - 
- I like 'vanilla' while I can edit the css file myself sure, I like using what the author/dev intended, my experience here was influenced by using yours on mobile
- when everything else in the world is finished/fixed or someone is bored, enhancements for using the UI on mobile with focus on canvas interactions
- Oh, you should come up with a distinct name for your UI to set it apart from auto, just a thought.
---------------


With regards to the theme: I am a weird kind of purist when it comes to a lot of things. I will add extensions all day, but I like to use themes and such that are "vanilla" that is, either OS default dark/light or what the author/developer intended. Minecraft for instance, I loved shaders, but not texture packs, aside from a guy that took the vanilla textures and gave them normal and specular maps which made it my go-to for years. 

No, don't change it just for me, but it came from a thought directly involved with my next passing thought -

And these are like, if someone is bored and feels like it suggestions, or everything else ever is done and the foundation of the UI is perfect and bug-free -

I spend a lot of time messing with SD on my phone and tt works well enough as-is but there are some pretty big areas for QoL improvements. I don't know if this is something gradio has to do themselves or if there's an avenue here for anyone to make gui changes to improve the mobile experience.

Common issues for me:
- theme makes it tougher to view on a smaller screen where tick boxes etc appear "bigger."
- tick boxes, sliders, etc could all use some kind of gui adjustment making it harder to accidentally move a value up or down or select a new sampler on accident while scrolling up and down.
- portrait/landscape considerations, I had a neat idea about supporting split screen mode, have the live preview sit in half and the other access to the gradio controls, dunno if you need to actually invoke android/ios native split screen scenario with a browser, but a function like that
- with considerations from suggestion above, it's definitely tough to do a plethora of things on mobile when interacting with the canvas in either inpaint, pose editors etc.

**Comment by vladmandic** [comment]
Re: Theme

Pretty much yes on all. When I started on CSS, it was just quick&dirty few fixes and then it grew and grew.
When I find the time, I need to get rid of all absolute sizing and convert to relative and that would take care of most of such things and also allow to increase/decrease size as user wanted. I have a feeling just cleaning it up to make it truly cross-portable would solve a lot of issues.

> No, don't change it just for me, but it came from a thought directly involved with my next passing thought -

The thing is, its not "just for you", i actively want suggestions. And some will make it :)

> Oh, you should come up with a distinct name for your UI to set it apart from auto, just a thought.

LOL. It was never my intention for this fork to blow up as it did, it was pretty much my private playground, so never gave branding any thought.

**Comment by myndxero** [comment]
Well there's suggestions, and then there's being nitpicky. Just making sure I don't come across that way.

Do a pole and take suggestions on subreddit or smthg. Can you change repo name without causing  issues to end users? That's the important part, if you want this to grow keeping it easy for users is most important. Course, even if you don't wish to become massive, I will stick around as long as active.

Never going back to auto1111 unless there's some new amazing thing I want to check out that he's done to set himself apart, but I'd just run both, tho i am using yours more and more exclusively.

**Reply by vladmandic** [reply]
my goal is that if repo gains popularity and i get other maintainers to join (i don't want it to rely purely on me) to move repo outside of a private account into org account with multiple admins - i guess that would be the good time to talk about branding.

atm, the only thing that would kill this repo is that if it becomes popular with new users, but nobody wants to join in to contribute.

**Comment by myndxero** [comment]
I am along for the ride. Wish I could help but I am so behind on stuff. I
learned C++ ages ago but barely remember it. I can stumble around style
sheet and html a bit, but that's. I'd say I could maybe offer something
graphically, but AI far eclipses my Photoshop skills tho there's precision
in PS that AI doesn't have the tools for yet by itself.

**Reply by vladmandic** [reply]
well, i'm an avid adobe lightroom guy for years, so i get it :)

**Comment by myndxero** [comment]
Oh just thought of something, seem to be missing the refresh button on
styles. Use it a lot on mobile especially cuz of times phone kicks the page
out of memory and when using latent couple.

**Reply by vladmandic** [reply]
never noticed, will take a look. but refresh should only be needed if you modify sytles.csv file outside of webui while webui is running.
anything else is actual issue and i'd rather fix it than hide the issue with refresh button.

**Comment by myndxero** [comment]
I do as I have it loaded in notepad Plus plus, plus when I use mobile and
it's running on my PC, restarting the UI isn't good enough to get it to
show up without the refresh button. Just a me thing

**Comment by myndxero** [comment]
<pre><code><del>Is there a way I can get the UI startup to ignore certain files? I've an update-all script I use for extensions and the UI crashes on it being in the folder. I know you can disable the extension security thingy as well but since I use --listen to connect remotely it's a thing.

If the current check you have in place is important to you, I absolutely do not mind extra steps on my end. Don't compromise how you run your UI on this issue.

Random side note, I used to use 'call update-all' in my webui.bat, but something somewhere changed and it no longer updates what's in extensions, only what's in the root folder. So weird, used to work fine. I am def a novice in writing these windows scripts.</del>
</code></pre>

I renamed it .placeholder.zip and it worked that way haha. Is it the whole word or simply '.' that makes it skip it?

**Reply by vladmandic** [reply]
i'm not sure i understand the last post.

**Reply by myndxero** [reply]
> i'm not sure i understand the last post.

I partially figured out the answer to my own question. I generally don't delete posts, but strikethrough them so they're still there as a reference.

I had a file in /extensions called update-all.bat, that I used with my auto1111 install. I called it with my custom webui-user.bat to automatically update all my downloaded extensions before loading the UI, first thing it did before teh webui call. I wanted to do that with yours, but your launcher would error.

![image](https://user-images.githubusercontent.com/113237671/233360041-37cf5d52-38bc-4b8a-ad6f-f69d9648d1f2.png)

Since I saw that .placeholder non-file, I decided to try renaming just to see if it would work as my original question was, "how can I get your launcher to ignore files in /extensions?"

Sorry I've not slept yet and am pretty scatterbrained, been all over reddit and github for hours.

**Reply by myndxero** [reply]
And reading more of the text in the error screenshot, I now see why, you have it auto update already. Prob tries to update my update-all.bat with a git pull or something. Derp.

**Reply by vladmandic** [reply]
yup, most likely...
